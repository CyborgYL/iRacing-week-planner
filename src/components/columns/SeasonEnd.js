// @flow

import * as React from 'react';
import moment from 'moment';
import { useTranslation } from 'react-i18next';

type Props = {
  race: {
    seriesEnd: moment.Moment,
  }
};

export default function SeasonEnd({ race }: Props) {
  const { t } = useTranslation();
  const date = moment(race.seriesEnd).local().toDate();
  return (
    <td>
      <div>
        {t('{{date, YYYY-MM-DD}}', { date })}
      </div>
    </td>
  );
}
