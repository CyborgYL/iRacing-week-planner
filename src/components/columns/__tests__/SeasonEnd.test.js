import moment  from 'moment';
import React from 'react';
import { shallow } from 'enzyme';
import { describe, test } from '@jest/globals';

import SeasonEnd from '../SeasonEnd';

describe('components/columns/SeasonEnd', () => {
  test('renders correctly', () => {
    const component = shallow(
      <SeasonEnd
        race={{
          seriesEnd: moment('2020-09-26T01:00:00.000Z').utc(),
        }}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.text()).toBe("2020-09-26");
  });
});
