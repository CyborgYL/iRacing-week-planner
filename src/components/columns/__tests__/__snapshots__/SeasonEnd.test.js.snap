// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/SeasonEnd renders correctly 1`] = `
<td>
  <div>
    2020-09-26
  </div>
</td>
`;
